backend/de.metas.externalsystem/src/main/java/de/metas/externalsystem/process/InvokeExternalSystemProcess.java

```diff
@@ -0,0 +1,159 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.externalsystem.process;
+
+import com.google.gson.Gson;
+import de.metas.common.externalsystem.ExternalSystemConstants;
+import de.metas.common.externalsystem.JsonExternalSystemName;
+import de.metas.common.externalsystem.JsonExternalSystemRequest;
+import de.metas.externalsystem.ExternalSystemChildConfig;
+import de.metas.externalsystem.ExternalSystemParentConfig;
+import de.metas.externalsystem.service.IExternalSystemConfigDAO;
+import de.metas.i18n.AdMessageKey;
+import de.metas.i18n.IMsgBL;
+import de.metas.organization.IOrgDAO;
+import de.metas.process.IADPInstanceDAO;
+import de.metas.process.IProcessDefaultParameter;
+import de.metas.process.IProcessDefaultParametersProvider;
+import de.metas.process.IProcessPrecondition;
+import de.metas.process.IProcessPreconditionsContext;
+import de.metas.process.JavaProcess;
+import de.metas.process.Param;
+import de.metas.process.ProcessInfo;
+import de.metas.process.ProcessPreconditionsResolution;
+import de.metas.rest_api.utils.MetasfreshId;
+import de.metas.util.Services;
+import lombok.NonNull;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpPut;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.springframework.http.MediaType;
+
+import javax.annotation.Nullable;
+import java.io.UnsupportedEncodingException;
+import java.sql.Timestamp;
+import java.util.HashMap;
+import java.util.Map;
+
+public abstract class InvokeExternalSystemProcess extends JavaProcess implements IProcessPrecondition, IProcessDefaultParametersProvider
+{
+	public final static AdMessageKey MSG_ERR_NO_EXTERNAL_SELECTION = AdMessageKey.of("NoExternalSelection");
+	public final static AdMessageKey MSG_ERR_MULTIPLE_EXTERNAL_SELECTION = AdMessageKey.of("MultipleExternalSelection");
+
+	public final IExternalSystemConfigDAO externalSystemConfigDAO = Services.get(IExternalSystemConfigDAO.class);
+	public final IADPInstanceDAO pInstanceDAO = Services.get(IADPInstanceDAO.class);
+
+	private static final String PARAM_CONFIG_ID = "configId";
+	@Param(parameterName = PARAM_CONFIG_ID)
+	protected MetasfreshId configId;
+
+	private static final String PARAM_SINCE = "since";
+	@Param(parameterName = PARAM_SINCE)
+	protected Timestamp since;
+
+	private static final String PARAM_EXTERNAL_REQUEST = "External_Request";
+	@Param(parameterName = PARAM_EXTERNAL_REQUEST)
+	protected String externalRequest;
+
+	private IOrgDAO orgDAO = Services.get(IOrgDAO.class);
+
+	@Override
+	protected String doIt() throws Exception
+	{
+
```

Comment:
please do addLog(...) to log as the call starts and ends( and what the statuscode was).
this will then end up in `AD_Pinstance_Log` and be a valuable info

Reply:
added logging